frame/session/src/lib.rs

```diff
@@ -651,7 +650,7 @@ impl<T: Trait> Module<T> {
 		}
 	}
 
-	fn load_keys(v: &T::ValidatorId) -> Option<T::Keys> {
+	pub fn load_keys(v: &T::ValidatorId) -> Option<T::Keys> {
```

Comment:
Not happy that I ended up doing this.

Reply:
What is going on here?

Reply:
I had to add some some direct calls into session module to get the keys of the validators. Currently there are two options to do this: 

- Use what @tomusdrw provides in `SubmitSignedTransaction` traits, but we ended in a loophole there and it ended up being too complicated for now.
- A module can implement `OneSessionHandler` and receive keys per session, and store them somewhere, similar to Im-online. This is bad since 1- needs an extra, duplicate storage in staking to keep the latest keys. 2- it is an overkill for staking since it is basically a feedback loop. Staking tells the authorities to session and then requests them back again? 

(although staking tells the IDs, staking does not know any key type)

Reply:
If we we end up with a separate (app-specific) keys for `staking` module, the `OneSessionHandler` approach will be necessary as that's where the new keys will be generated. IMHO that's the way we should go with.

Reply:
Okay let's go with app-specific key then.